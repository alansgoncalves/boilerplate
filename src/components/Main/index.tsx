import * as S from './styles'

const Main = ({
  title = 'React avançado',
  description = 'Typescript, ReactJs, NextJs, Styled components'
}) => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Imagem de um átomo e React avançado escrito ao lado"
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration
      src="/img/hero-illustration.svg"
      alt="Pessoa desenvolvedora de frente para uma tela com código"
    />
  </S.Wrapper>
)

export default Main
