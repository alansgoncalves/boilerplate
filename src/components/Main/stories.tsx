import { Meta, Story } from '@storybook/react'
import { withKnobs, text } from '@storybook/addon-knobs'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  decorators: [withKnobs]
} as Meta

export const Basic: Story = () => (
  <Main
    title={text('Title', 'React avanÃ§ado')}
    description={text(
      'Description',
      'Typescript, ReactJs, NextJs, Styled components'
    )}
  />
)
