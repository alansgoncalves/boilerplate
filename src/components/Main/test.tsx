import { render, screen } from '@testing-library/react'

import Main from '../Main'

describe('Testa o componente Main', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)
    expect(
      screen.getByRole('heading', { name: /React avan√ßado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
