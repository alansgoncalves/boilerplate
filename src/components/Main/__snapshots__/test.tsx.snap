// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testa o componenete Main should render the heading 1`] = `
<div>
  <h1>
    React avan√ßado
  </h1>
</div>
`
